string name
---
string[] names
float64[] centers
int64 num